sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -103,12 +123,53 @@ protected String visitIsNotNull(String v) {
     return v + " IS NOT NULL";
   }
 
-  protected String visitBinaryComparison(String name, String l, String r) {
-    return "(" + l + ") " + name + " (" + r + ")";
+  protected String visitStringPredicate(String name, GeneralScalarExpression strPredicate) {
+    String l = build(strPredicate.children()[0]);
+    String r = build(strPredicate.children()[1]);
+    String value = r.replaceAll("'", "");
+    switch (name) {
+      case "STARTS_WITH":
+        return l + " LIKE '" + value + "%'";
+      case "ENDS_WITH":
+        return l + " LIKE '%" + value + "'";
+      case "CONTAINS":
+        return l + " LIKE '%" + value + "%'";
+      default:
+        return visitUnexpectedExpr(strPredicate);
+    }
   }
 
-  protected String visitBinaryArithmetic(String name, String l, String r) {
-    return "(" + l + ") " + name + " (" + r + ")";
+  protected String visitBinaryComparison(
+    String name, String l, int lChildNum, String r, int rChildNum) {
+    switch (name) {
+      case "<=>":
+        return "(" + l + " = " + r + ") OR (" + l + " IS NULL AND " + r + " IS NULL)";
+      default:
+        String left = l, right = r;
+        if (lChildNum > 1) {
+          left = "(" + l + ")";
+        }
+        if (rChildNum > 1) {
+          right = "(" + r + ")";
+        }
+        return left + " " + name + " " + right;
+    }
+  }
+
+  private int childNum(Expression expr) {
+    return expr.children().length;
+  }
+
+  protected String visitBinaryArithmetic(
+    String name, String l, int lChildNum, String r, int rChildNum) {
+    String left = l, right = r;
+    if (lChildNum > 1) {
```

Comment:
Remember: we expect users to override `visitXXX` methods and we should try out best to not push the work to the visit methods. We can add parentheses at the caller side
```
private String inputToSQL(Expression input) {
  if (input.children().length > 1) {
    return "(" + build(input) + ")";
  } else {
    return build(input);
  }
}
...
return visitBinaryArithmetic(name, inputToSQL(e.children()[0]), inputToSQL(e.children()[1]));
```